### PR TITLE
Set outbound interface, like `ping -I` option

### PR DESCRIPTION
**Pull request by ilolicon**
Hello,

I attempted to implement a useful feature - sending data packets from a specific network interface, similar to what the "ping" command's "-I" option does in Linux.
Hoping I did it correctly. Test successfully passed on Linux OS.

Thanks!

The issues related to this feature:

go-ping/ping#111

go-ping/ping#204

### CODE REVIEW

**Comment by SuperQ** [comment]
This needs a DCO sign-off. You can use `git commit -s --amend` to add it.

File: ping.go [hunk 1]
```diff
@@ -182,6 +182,9 @@ type Pinger struct {
 	// Source is the source IP address
 	Source string
 
+	// Iface used to send/recv ICMP messages
+	Iface string
```
Comment (Hipska) [inline ping.go:186]: I don't like the short name
```suggestion
	// Interface used to send/recv ICMP messages
	Interface string
```

**Review by Hipska — APPROVED** [review]

File: ping.go (hunk 1 above)
Comment (ilolicon) [reply ping.go]: The original intention was just to avoid confusion with the go `interface` keyword

Comment (Hipska) [reply ping.go]: Maybe `InterfaceName` then if needed?

**Review by Hipska — APPROVED** [review]

File: cmd/ping/ping.go [hunk 2]
```diff
@@ -29,6 +29,9 @@ Examples:
     # ping google for 10 seconds
     ping -t 10s www.google.com
 
+    # ping google specified interface
+    ping -I eth1 www.goole.com
```
Comment (ja-he) [inline cmd/ping/ping.go:33]: there is a typo here: `goo[g]le.com`

**Comment by bluecmd** [comment]
FWIW, I would like to see this feature added in order to use VRF with this library.

**Comment by HsimWong** [comment]
I need this feature really bad. Can anyone merge this?

**Comment by bluecmd** [comment]
The PR needs to be rebased if it is to have any chance of getting merged.

File: cmd/ping/ping.go (hunk 2 above)
Comment (Hipska) [reply cmd/ping/ping.go:33]:
```suggestion
    ping -I eth1 www.google.com
```

**Comment by dan-sherwin** [comment]
Checking on this PR?  I could use it.  Is it dead?

**Comment by liuweiky** [comment]
Hi, any updates on this PR? It is quite useful when debugging the network on hosts with multiple net interfaces.

**Comment by MatthieuCoder** [comment]
I need this to use link-local IPv6 addresses

**Comment by Hipska** [comment]
At least the PR has conflicts, @ilolicon can you resolve them?

**Comment by ilolicon** [comment]
> At least the PR has conflicts, @ilolicon can you resolve them?

Sorry, I was too busy recently and didn't pay attention to the news. It seems that @MatthieuCoder has resolved the conflict.

**Pull request closed without merging**